pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -462,6 +467,74 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileBuildRun) updateBuildRunUsingTaskRunCondition(ctx context.Context, buildRun *buildv1alpha1.BuildRun, taskRun *v1beta1.TaskRun, trCondition *apis.Condition) error {
+	var reason, message string = trCondition.Reason, trCondition.Message
+
+	switch v1beta1.TaskRunReason(reason) {
+	case v1beta1.TaskRunReasonTimedOut:
+		reason = "BuildRunTimeout"
+		message = fmt.Sprintf("BuildRun %s failed to finish within %s",
+			taskRun.GetLabels()[buildv1alpha1.LabelBuildRun],
+			taskRun.Spec.Timeout.Duration,
+		)
+
+	case v1beta1.TaskRunReasonFailed:
+		if taskRun.Status.CompletionTime != nil {
+			var pod corev1.Pod
+			if err := r.client.Get(ctx, types.NamespacedName{Namespace: taskRun.Namespace, Name: taskRun.Status.PodName}, &pod); err != nil {
+				return err
+			}
+
+			// Since the container status list is not sorted, as a quick workaround mark all failed containers
+			var failures = make(map[string]struct{})
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode != 0 {
+					failures[containerStatus.Name] = struct{}{}
+				}
+			}
+
+			// Find the first container that failed
+			var failedContainer *corev1.Container
+			for i, container := range pod.Spec.Containers {
+				if _, has := failures[container.Name]; has {
+					failedContainer = &pod.Spec.Containers[i]
+					break
+				}
+			}
+
+			if failedContainer != nil {
+				message = fmt.Sprintf("buildrun step failed in pod %s, for detailed information: kubectl --namespace %s logs %s --container=%s",
+					pod.Name,
+					pod.Namespace,
+					pod.Name,
+					failedContainer.Name,
+				)
+			} else {
+				message = fmt.Sprintf("buildrun failed due to an unexpected error in pod %s: for detailed information: kubectl --namespace %s logs %s",
+					pod.Name,
+					pod.Namespace,
+					pod.Name,
+				)
```

Comment:
Given we know that the pod will have more than one container and that kubectl expects a container name if there is more than one, what about appending the `--all-containers` argument?

Reply:
totally, will add this.